librz/core/canalysis.c

```diff
@@ -6384,3 +6384,36 @@ RZ_API ut64 rz_core_prevop_addr_force(RzCore *core, ut64 start_addr, int numinst
 	}
 	return start_addr;
 }
+
+RZ_API void rz_core_perform_auto_analysis(RZ_NONNULL RzCore *core, RzCoreAnalysisType type) {
+	rz_return_if_fail(core);
+
+	ut64 timeout = rz_config_get_i(core->config, "analysis.timeout");
+	char *debugger = NULL;
+	ut64 old_offset = core->offset;
+	const char *notify = "Analyze all flags starting with sym. and entry0 (aa)";
+	rz_core_notify_begin(core, "%s", notify);
+	rz_cons_break_push(NULL, NULL);
+	rz_cons_break_timeout(timeout);
+	rz_core_analysis_all(core);
+	rz_core_notify_done(core, "%s", notify);
+	rz_core_task_yield(&core->tasks);
+	if (type == RZ_CORE_ANALYSIS_SIMPLE || rz_cons_is_breaked()) {
+		goto finish;
+	}
+	// Run pending analysis immediately after analysis
+	// Usefull when running commands with ";" or via rizin -c,-i
+	debugger = core->dbg->cur ? strdup(core->dbg->cur->name) : strdup("esil");
```

Comment:
You can move definition of the variable here, no need to define everything at the beginning of the function - we support only >=C99 standard.

Reply:
but `char *debugger = NULL;` at line 6398, so it should not.

Reply:
Don't forget this please